appender.go

```diff
@@ -153,33 +125,19 @@ func New(client elastictransport.Interface, cfg Config) (*Appender, error) {
 	if err != nil {
 		return nil, err
 	}
-	available := make(chan *BulkIndexer, cfg.MaxRequests)
-	for i := 0; i < cfg.MaxRequests; i++ {
-		bi, err := NewBulkIndexer(BulkIndexerConfig{
-			Client:                client,
-			MaxDocumentRetries:    cfg.MaxDocumentRetries,
-			RetryOnDocumentStatus: cfg.RetryOnDocumentStatus,
-			CompressionLevel:      cfg.CompressionLevel,
-			Pipeline:              cfg.Pipeline,
-			RequireDataStream:     cfg.RequireDataStream,
-			IncludeSourceOnError:  cfg.IncludeSourceOnError,
-		})
-		if err != nil {
-			return nil, fmt.Errorf("error creating bulk indexer: %w", err)
-		}
-		available <- bi
-	}
-	if cfg.Logger == nil {
-		cfg.Logger = zap.NewNop()
+	if err := BulkIndexerConfigFrom(cfg).Validate(); err != nil {
+		return nil, fmt.Errorf("error creating bulk indexer: %w", err)
 	}
 	indexer := &Appender{
+		pool:      cfg.BulkIndexerPool,
 		config:    cfg,
-		available: available,
+		client:    client,
+		id:        randStringBytes(32),
 		closed:    make(chan struct{}),
 		bulkItems: make(chan BulkIndexerItem, cfg.DocumentBufferSize),
 		metrics:   ms,
 	}
-	indexer.addUpDownCount(int64(len(available)), &indexer.availableBulkRequests, ms.availableBulkRequests)
+	indexer.addUpDownCount(int64(cfg.MaxRequests), &indexer.availableBulkRequests, ms.availableBulkRequests)
```

Comment:
IIUC this means that every Appender will increment by MaxRequests, which I don't think is correct? Should we perhaps move this to the BulkIndexerPool?

Reply:
I think it'd be best to keep this here for the Appender and perhaps introduce a new metric (that doesn't depend on legacy metrics) in the `BulkIndexerPool`.